sw/airborne/firmwares/rotorcraft/autopilot_arming_yaw.h

```diff
@@ -123,6 +127,11 @@ static inline void autopilot_arming_check_motors_on(void)
         break;
       case STATUS_MOTORS_AUTOMATICALLY_OFF: // Motors were disarmed externally
         //(possibly due to crash)
+        // If the vehicle was killed accidentally, allow rapid re-arm
+        if ( (get_sys_time_float() - motor_kill_time) < 10.0) {
+          autopilot.motors_on = true;
+          autopilot_check_motor_status = STATUS_MOTORS_ON;
+        }
         //wait extra delay before enabling the normal arming state machine
         autopilot.motors_on = false;
```

Comment:
motors_on seems to be reset to false even if the kill-time was < 10s.

Reply:
Good point, it will now only be re-armed the next iteration when it is in STATUS_MOTORS_ON... Which also works. I can put an else case though